src/auto.ts

```diff
@@ -170,13 +182,30 @@ export async function autoload(address: string, config: AutoloadConfig): Promise
     };
 
     if (result.proxies.length === 1 && result.proxies[0] instanceof DiamondProxyResolver) {
-        onProgress("loadDiamondFacets", { address });
         const diamondProxy = result.proxies[0] as DiamondProxyResolver;
-        const f = await diamondProxy.facets(provider, address);
-        Object.assign(facets, f);
+        if (config.followProxies) {
+            onProgress("loadDiamondFacets", { address });
+            const f = await diamondProxy.facets(provider, address);
+            Object.assign(facets, f);
+        } else {
+            result.followProxies = async function(selector?: string): Promise<AutoloadResult> {
+                if (selector) {
+                    // Follow a single selector for DiamondProxy
+                    onProgress("followProxies", { resolver: diamondProxy, address });
+                    const resolved = await diamondProxy.resolve(provider, address, selector);
+                    if (resolved !== undefined) return await autoload(resolved, config);
+                }
+                // Resolve all facets, unfortunately this requires doing the whole thing again with followProxy
+                // FIXME: Can we improve this codeflow easily?
+                // We could override the privider with a cached one here, but might be too magical and cause surprising bugs?
+                return await autoload(address, Object.assign({}, config, { followProxies: true }));
```

Comment:
@yohamta What do you think of this if there is a single DiamondProxy but followProxies is false?

I'm not too happy with it because it'll rerun `getCode` again, but not sure if this is an edge case that is important?

Reply:
Thank you! It looks good to me. As you mentioned, the redundant execution of getCode doesn’t seem too critical in this case. 

One thing I've been thinking about is that, for DiamondProxy, the `resolve()` method requires a `selector` argument. This makes it seem a bit challenging to retrieve the implementation addresses for the `facets` from user side. What are your thoughts on this? When `followProxy: false` is set and facets aren't being tracked, it would be helpful if users could easily get a list of facet addresses. Another potential solution could be to automatically resolve facets for DiamondProxy, even when `followProxy: false` is used.

Reply:
> One thing I've been thinking about is that, for DiamondProxy, the resolve() method requires a selector argument.

Correct, with this fix if you want to get all possible selectors, you'd need to do something like:

```ts
const result = whatsabi.autoload(address, { provider, followProxies: false });
const p = result.proxies[0] as DiamondProxyResolver;
const facets = await diamondProxy.facets(provider, address); // All possible address -> selector[] mappings
```

But just doing `followProxies()` should automagically do that for you and look them up and return the abi as it used to.

> Another potential solution could be to automatically resolve facets for DiamondProxy, even when followProxy: false is used.

This was the original behaviour we're trying to fix, right?

Reply:
Thank you for the code snippet, it's very helpful!

> But just doing `followProxies()` should automagically do that for you and look them up and return the abi as it used to.

Yes, that's true. On the other hand, some user might want to retrieve `ContractResult` for each facet. To do so, they can follow the method you've described above, I think.

> This was the original behaviour we're trying to fix, right?
Yes, exactly. I’m still unsure of the best way to handle DiamondProxy in this context, which is why I brought it up.

I’m also a bit curious if there could be cases where a deeper hierarchy of proxies exists under the DiamondProxy facets.

Reply:
> Thank you for the code snippet, it's very helpful!

I'll add it to the docs! Thanks for the prompt. :)

> On the other hand, some user might want to retrieve ContractResult for each facet individually. In that case, they can follow the method you've described above, which I think works well.

Hmm are individual facet addresses typically verified? We should look at some examples of DiamondProxies in the wild to see how they're treated.

> Yes, exactly. I’m still unsure of the best way to handle DiamondProxy in this context, which is why I brought it up.

That's fair. I'd like to do a release today or tomorrow, so I might just merge the ContractResult work and revert the changes to the followProxies for now, and we can think about it more after the release. :)

> I’m also a bit curious if there could be cases where a deeper hierarchy of proxies exists under the DiamondProxy facets.

That's a good question! I'm skeptical this is used in practice, but it's certainly possible to do!

Someone should deploy a weird fractal multi-proxy monstrosity just so we have something to test against. 😅

Reply:
Sounds great, and thank you very much! Looking forward to the next release.